sourced/ml/transformers/content2ids.py

```diff
@@ -26,94 +28,84 @@ def __init__(self, **options):
         def format(self, tokensource, outfile):
             self.callback(tokensource)
 
-    def __init__(self, args, documents_column, **kwargs):
+    def __init__(self, language_mapping: Dict, column_names: NamedTuple,
+                 split: bool, idfreq: bool, **kwargs):
         super().__init__(**kwargs)
-        self.documents_column = documents_column
-        self.linguist2pygments = {}
-        self.split = args.split
-        self.idfreq = args.idfreq
-        self.output = args.output
+        self.column_names = column_names
+        self.linguist2pygments = language_mapping
+        self.split = split
+        self.idfreq = idfreq
 
-    def __call__(self, rows):
+    def __call__(self, rows: RDD):
         list_RDDs = []
-        self.build_mapping()
-        processed = rows.flatMap(self._process_row).persist()
-
+        processed = rows.flatMap(self._process_row)
```

Comment:
And this is the problem because you will calculate `self_process_row` every time. That's why I suggested to split this class and put a Cacher in the middle.